Panic when serverVersion is not defined at Cluster level

**What happened?**

Due to merging this [pull request](https://github.com/k8ssandra/k8ssandra-operator/pull/934), the k8ssandra-operator panics when `serverVersion` is not defined at the Cluster level (`spec.cassandra.serverVersion`). It also doesn't check `serverVersion` at the Datacenter level (`spec.cassandra.datacenter[n].serverVersion`).

**Did you expect to see something different?**

Correct checking both configuration(dc and cluster)

**How to reproduce it (as minimally and precisely as possible):**

Define `serverVersion` on the Datacenter-level configuration, but don't provide `serverVersion` at the Cluster level.

**Environment**

- K8ssandra Operator version: 1.8.0
- Kubernetes version information: 1.26.3
- Kubernetes cluster kind: self-managed

* K8ssandra Operator Logs:

```
2023-07-16T04:55:52.158Z	INFO	Observed a panic in reconciler: Invalid Semantic Version	{"controller": "k8ssandracluster", "controllerGroup": "k8ssandra.io", "controllerKind": "K8ssandraCluster", "K8ssandraCluster": {"name":"damir-test","namespace":"default"}, "namespace": "default", "name": "damir-test", "reconcileID": "9faecfa6-d79e-4aa7-b6a9-47731534652b"}
panic: Invalid Semantic Version [recovered]
	panic: Invalid Semantic Version

goroutine 740 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x183a420, 0xc000114740})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/Masterminds/semver/v3.MustParse(...)
	/go/pkg/mod/github.com/!masterminds/semver/v3@v3.1.1/version.go:214
github.com/k8ssandra/k8ssandra-operator/pkg/telemetry.IsNewMetricsEndpointAvailable({0x0?, 0x0?})
	/workspace/pkg/telemetry/validation.go:47 +0x125
github.com/k8ssandra/k8ssandra-operator/controllers/k8ssandra.(*K8ssandraClusterReconciler).createDatacenterConfigs(0xc0005347d0, {0x1d24ba8, 0xc0003dae70}, 0xc000c1e4e0, {{0x1d26fe0?, 0xc0003daed0?}, 0x0?}, 0x20000c000c38ce8?)
	/workspace/controllers/k8ssandra/dcconfigs.go:87 +0xabd
github.com/k8ssandra/k8ssandra-operator/controllers/k8ssandra.(*K8ssandraClusterReconciler).reconcileDatacenters(0xc0005347d0, {0x1d24ba8, 0xc0003dae70}, 0xc000c1e4e0, {{0x1d26fe0?, 0xc0003daed0?}, 0x0?})
	/workspace/controllers/k8ssandra/datacenters.go:45 +0x18e
github.com/k8ssandra/k8ssandra-operator/controllers/k8ssandra.(*K8ssandraClusterReconciler).reconcile(0xc000c1e4e0?, {0x1d24ba8, 0xc0003dae70}, 0xc000c1e4e0, {{0x1d26fe0?, 0xc0003daed0?}, 0xc000b3b360?})
	/workspace/controllers/k8ssandra/k8ssandracluster_controller.go:143 +0x2e5
github.com/k8ssandra/k8ssandra-operator/controllers/k8ssandra.(*K8ssandraClusterReconciler).Reconcile(0xc0005347d0, {0x1d24ba8, 0xc0003dae70}, {{{0xc000b3b360, 0x7}, {0xc000b3b350, 0xa}}})
	/workspace/controllers/k8ssandra/k8ssandracluster_controller.go:91 +0x314
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1d24ba8?, {0x1d24ba8?, 0xc0003dae70?}, {{{0xc000b3b360?, 0x17de100?}, {0xc000b3b350?, 0x1d127c8?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000366a00, {0x1d24b00, 0xc0004f1360}, {0x18c93a0?, 0xc000116920?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000366a00, {0x1d24b00, 0xc0004f1360})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x587```

--- a/CHANGELOG/CHANGELOG-1.9.md
+++ b/CHANGELOG/CHANGELOG-1.9.md
@@ -17,3 +17,4 @@ When cutting a new release, update the `unreleased` heading to the tag being gen
 
 * [BUGFIX] [#981](https://github.com/k8ssandra/k8ssandra-operator/issues/981) Fix race condition in K8ssandraTask status update
 * [DOCS] [#1019](https://github.com/k8ssandra/k8ssandra-operator/issues/1019) Add PR review guidelines documentation
+* [BUGFIX] [#1023](https://github.com/k8ssandra/k8ssandra-operator/issues/1023) Ensure merged serverVersion passed to IsNewMetricsEndpointAvailable()

--- a/controllers/k8ssandra/dcconfigs.go
+++ b/controllers/k8ssandra/dcconfigs.go
@@ -84,7 +84,7 @@ func (r *K8ssandraClusterReconciler) createDatacenterConfigs(
 		// The new metrics endpoint is available since 3.11.13 and 4.0.4.
 		// If MCAC is disabled and the new metrics endpoint is not available then we should return an error.
 		mergedTelemetrySpec := MergeTelemetrySpecs(kc, dcTemplate)
-		if !mergedTelemetrySpec.IsMcacEnabled() && !telemetry.IsNewMetricsEndpointAvailable(kc.Spec.Cassandra.ServerVersion) && kc.Spec.Cassandra.ServerType == k8ssandraapi.ServerDistributionCassandra {
+		if !mergedTelemetrySpec.IsMcacEnabled() && !telemetry.IsNewMetricsEndpointAvailable(dcConfig.ServerVersion.String()) && kc.Spec.Cassandra.ServerType == k8ssandraapi.ServerDistributionCassandra {
 			return dcConfigs, errors.New("new metrics endpoint is only available since Cassandra 3.11.13/4.0.4, so MCAC cannot be disabled")
 		} else {
 			logger.Info("new metrics endpoint is available, so MCAC can be disabled", "serverVersion", kc.Spec.Cassandra.ServerVersion)

--- a/test/testdata/fixtures/single-dc/k8ssandra.yaml
+++ b/test/testdata/fixtures/single-dc/k8ssandra.yaml
@@ -4,7 +4,6 @@ metadata:
   name: test
 spec:
   cassandra:
-    serverVersion: 4.0.6
     clusterName: "Weird Cluster Name"
     telemetry:
       mcac:
@@ -30,6 +29,7 @@ spec:
           name: dc1
         k8sContext: kind-k8ssandra-0
         size: 3
+        serverVersion: 4.0.6
         storageConfig:
           cassandraDataVolumeClaimSpec:
             storageClassName: standard

